
# HG changeset patch
# User Elliott Sales de Andrade <qulogic@pidgin.im>
# Date 1392341598 18000
# Node ID 922e7abddb4dbc1a397aad9c1d3b6fd091c4e392
# Parent  a591c0486b041e029075e12b446430cf32b27d24
Fix call to Py_SetProgramName in Python3.

Refs #15969.

diff --git a/finch/libgnt/gntwm.c b/finch/libgnt/gntwm.c
--- a/finch/libgnt/gntwm.c
+++ b/finch/libgnt/gntwm.c
@@ -1577,7 +1577,17 @@
 	gnt_bindable_class_register_action(GNT_BINDABLE_CLASS(klass), "run-python", run_python,
 				GNT_KEY_F3, NULL);
 	if (!Py_IsInitialized()) {
+#if PY_MAJOR_VERSION >= 3
+		wchar_t *name;
+		size_t len;
+		len = mbstowcs(NULL, "gnt", 0);
+		name = g_new(wchar_t, len + 1);
+		mbstowcs(name, "gnt", len + 1);
+		Py_SetProgramName(name);
+		g_free(name);
+#else
 		Py_SetProgramName("gnt");
+#endif
 		Py_Initialize();
 		started_python = TRUE;
 	}

